server/iterable_weakmap.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * @license
+ * IterableWeakMap
+ *
+ * Copyright 2018 Google Inc.
+ * https://github.com/NeilFraser/CodeCity
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview A WeakMap that's iterable.
+ * @author cpcallen@google.com (Christohper Allen)
+ */
+'use strict';
+
+const weak = require('weak');
+
+/**
+ * Function to clean up dead cells from an IterableWeakMap when a key
+ * object has been garbaged collected.
+ * @this {!IterableWeakMap} IterableWeakMap to remove GCed key from.
+ * @param {!Cell} cell Cell containing GCed key.
+ * @return {void}
```

Comment:
Remove this?

Reply:
I'm going to leave this, for consistency with the official externs for Map (and with some of the declarations earlier in this file.)

I agree that it it is interesting that this is so infrequently used in google3, though.